apiserver/remoterelations/state.go

```diff
@@ -174,10 +184,47 @@ type Application interface {
 	Endpoints() ([]state.Endpoint, error)
 }
 
+// Address provides access to an address in global state.
+type Address interface {
+	// SubnetCIDR returns the CIDR of the subnet this IP address comes from.
+	SubnetCIDR() string
+}
+
+type statePoolShim struct {
+	*state.StatePool
+}
+
+func (pool statePoolShim) Get(modelUUID string) (RemoteRelationsState, func(), error) {
+	st, closer, err := pool.StatePool.Get(modelUUID)
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+	return stateShim{st}, closer, nil
+}
+
 type stateShim struct {
 	*state.State
 }
 
+func (st stateShim) AllIPAddresses() (addresses []Address, err error) {
+	addr, err := st.State.AllIPAddresses()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	for _, a := range addr {
+		addresses = append(addresses, addressShim{a})
+	}
+	return addresses, nil
+}
+
+type addressShim struct {
```

Comment:
We need the shim because state.Address has provide attributes that would need to be set